### PR TITLE
RUBY 2625 wcr back office lower tier registrations are shown in conviction check work flow

### PR DESCRIPTION
**Pull request by jjromeo**
- [RUBY-2625] Add raketask to clear conviction sign-offs for lower tier registrations
- [RUBY-2625] Rake task will correctly destroy conviction sign offs for a lower tier registration

[RUBY-2625]: https://eaflood.atlassian.net/browse/RUBY-2625?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RUBY-2625]: https://eaflood.atlassian.net/browse/RUBY-2625?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/tasks/one_off/clear_lower_tier_with_conviction_flags.rake [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      registration.conviction_sign_offs.destroy_all
+      registration.save
+    end
+  end
+
+  def scope
+    WasteCarriersEngine::Registration.collection.aggregate(lower_tier_conviction_pipeline).pluck(:_id)
+  end
+
+  def lower_tier_conviction_pipeline
```
Comment (PaulDoyle-DEFRA) [inline lib/tasks/one_off/clear_lower_tier_with_conviction_flags.rake:18]: Can this be done using a single `where` clause with `tier: "LOWER"` and `convictionSignoffs.exists: true` ?

File: lib/tasks/one_off/clear_lower_tier_with_conviction_flags.rake [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+namespace :one_off do
+  desc "Clear LOWER tier registrations with conviction flags"
+  task clear_lower_tier_with_conviction_flags: :environment do
+    registration_ids = scope
+    registration_ids.each do |id|
```
Comment (PaulDoyle-DEFRA) [inline lib/tasks/one_off/clear_lower_tier_with_conviction_flags.rake:7]: Could this be done using `update_all` and setting `conviction_sign_offs` to `[]` ?

File: lib/tasks/one_off/clear_lower_tier_with_conviction_flags.rake (hunk 1 above)
Comment (jjromeo) [reply lib/tasks/one_off/clear_lower_tier_with_conviction_flags.rake:18]: This doesn't seem to work, I've tried 

```
    WasteCarriersEngine::Registration.where(
      tier: "LOWER",
      convictionSignOffs: { "$exists": true, "$ne": [] }
    ).pluck(:_id)
```
and

```
    WasteCarriersEngine::Registration.where(
      tier: "LOWER",
      :convictionSignOffs.ne => [],
      :convictionSignOffs.exists => true
    ).pluck(:_id)
```

File: lib/tasks/one_off/clear_lower_tier_with_conviction_flags.rake (hunk 2 above)
Comment (jjromeo) [reply lib/tasks/one_off/clear_lower_tier_with_conviction_flags.rake:7]: `update_all` seems to only be useable with the activerecord query methods, which I can't get to work in this specific case

**Review by PaulDoyle-DEFRA — APPROVED** [review]

**Pull request merged**